src/client/client.py

```diff
@@ -17,197 +19,203 @@ class Client:
     USER_CONF = Path('client/_data/config/user.conf')
+    EOM = '\n' # End of Message sign
 
 
     @staticmethod
-    def upload(file_path: str):
+    async def upload(file_path: str):
         """Upload file to a server."""
 
         if os.path.exists(file_path):
-            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as client:
-                try:
-                    client.connect((Client.SERVER_IP, Client.SERVER_PORT))
-                except ConnectionRefusedError as e:
-                    print(e)
-                    print("Server probably isn't running!!!")
-                    exit(1)
-
-                file_name = os.path.basename(file_path)
-
-                # Send UPLOAD request
-                client.send(f"UPLOAD;{file_name};{Client.get_username()}".encode())
-
-                # Receive aswer
-                answer = client.recv(Client.BUFFER_SIZE).decode()
-
-                if 'OK' in answer:
-                    # Answer: 'OK;Authenticated'
-
-                    file_size = os.path.getsize(file_path)
-                    progress = tqdm.tqdm(range(file_size), f"Sending {file_name}", unit="B", unit_scale=True, unit_divisor=1024)
-
-                    # Send file
-                    with open(file_path, "rb") as f:
-                        while True:
-                            bytes_read = f.read(Client.BUFFER_SIZE)
-                            if not bytes_read:
-                                break
-                            client.sendall(bytes_read)
-                            progress.update(len(bytes_read))
-
-                elif 'NotAuthenticated' in answer:
-                    # Answer: 'ERROR;NotAuthenticated'
-                    print(f"User '{Client.get_username()}' can't be authenticated, CANNOT upload")
-                else:
-                    # Answer: Any
-                    print("client.py: Unknown exeption")
-            
-            client.close()
-
-        else:
-            # Client FileNotFoundError
-            print(f"File '{file_path}' can't be reached, No action taken...")
-            print(f"Check correct file name and path and try again ")
-
-
-    @staticmethod
-    def download(file_name: str):
-        """Download file from a server."""
-
-        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as client:
             try:
-                client.connect((Client.SERVER_IP, Client.SERVER_PORT))
+                reader, writer = await asyncio.open_connection(Client.SERVER_IP, Client.SERVER_PORT)
             except ConnectionRefusedError as e:
                 print(e)
                 print("Server probably ins't running!!!")
                 exit(1)
+ 
+            file_name = os.path.basename(file_path)
 
-            # Send DOWNLOAD request
-            client.send(f"DOWNLOAD;{file_name};{Client.get_username()}".encode())
+            # Send UPLOAD request
+            writer.write(f"UPLOAD;{file_name};{Client.get_username()}{Client.EOM}".encode())
```

Comment:
Dívám, že jsi našel funkci `get_username()` co jsem ti na to připravil, to je fajn.